nucypher/keystore/keystore.py

```diff
@@ -145,9 +158,24 @@ def del_policy_arrangement(self, arrangement_id: bytes, session=None):
         session.query(PolicyArrangement).filter_by(id=arrangement_id).delete()
         session.commit()
 
+    def del_expired_policy_arrangements(self, session=None, now=None) -> int:
+        """
+        Deletes all expired PolicyArrangements from the Keystore.
+        """
+        session = session or self._session_on_init_thread
+        now = now or datetime.now()
+        result = session.query(PolicyArrangement).filter(PolicyArrangement.expiration < now)
+        deleted_records = result.delete()
+        try:
+            session.commit()
+        except OperationalError:
+            session.rollback()
```

Comment:
Should we add some logging here in the event that a rollback occurs?

Reply:
Maybe this should raise some sort of exception for `__prune_arrangements` to handle?

Reply:
:heavy_check_mark: